src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const actionsArray = [];
+  const resultObject = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(resultObject, action.extraData, actionsArray);
+        break;
+      case 'removeProperties':
+        removeProperties(resultObject, action.keysToRemove, actionsArray);
+        break;
+      case 'clear':
+        clear(actionsArray, resultObject);
+        break;
+    }
```

Comment:
There is no default case in the switch statement. It's a good practice to include a default case for handling unexpected action types.